core/src/main/java/org/infinispan/configuration/parsing/Parser.java

```diff
@@ -1743,19 +1846,25 @@ protected void parseEviction(XMLExtendedStreamReader reader, ConfigurationBuilde
             }
             case MAX_ENTRIES: {
                log.evictionMaxEntriesDeprecated();
-               builder.eviction().size(Long.parseLong(value));
+               long parsedLong = Long.parseLong(value);
+               builder.eviction().size(parsedLong);
+               builder.memory().size(parsedLong);
                break;
             }
             case THREAD_POLICY: {
                builder.eviction().threadPolicy(EvictionThreadPolicy.valueOf(value));
                break;
             }
             case TYPE: {
-               builder.eviction().type(EvictionType.valueOf(value));
+               EvictionType type = EvictionType.valueOf(value);
+               builder.eviction().type(type);
+               builder.memory().evictionType(type);
                break;
             }
             case SIZE: {
-               builder.eviction().size(Long.parseLong(value));
+               long size = Long.parseLong(value);
+               builder.eviction().size(size);
```

Comment:
Instead of invoking both, wouldn't it be better to have the eviction stuff merely act as a facade to the memory configuration